llvm/lib/AsmParser/LLLexer.cpp

```diff
@@ -984,9 +987,16 @@ lltok::Kind LLLexer::LexIdentifier() {
 ///    HexPPC128Constant 0xM[0-9A-Fa-f]+
 ///    HexHalfConstant   0xH[0-9A-Fa-f]+
 ///    HexBFloatConstant 0xR[0-9A-Fa-f]+
+///    FIXME: I have temporarily added these prefixes temporarly to mimic
+///    the DFP attributes in C++. But this is just a placeholder for
+///    the real prefix.
+///    HexDecimal32Constant 0xSD[0-9A-Fa-f]+
+///    HexDecimal64Constant 0xDD[0-9A-Fa-f]+
```

Comment:
The `0xDD` prefix looks like it introduces an ambiguity with the existing `HexFPConstant`. The literal suffixes (`DF`, `DD`, and `DL`) have the same problem.

It looks like we can probably omit this support; the C23 draft, in 6.4.4.2, Floating constants, p2, states that DFP suffixes shall not be used with hexadecimal floating constants.

Reply:
Ok. So this can be removed?

Reply:
I think so, yes.